File: programming/problem_6/problem_6.ts
function averageExcludingMinMax(inputNumbers: number[]): number {
  const min = Math.min(...inputNumbers);
  const max = Math.max(...inputNumbers);
  const filteredNumbers = inputNumbers.filter((e) => e !== min && e !== max);
  const sum = filteredNumbers.reduce((prev, curr) => prev + curr, 0);
  return sum / filteredNumbers.length;
}

console.log('averageExcludingMinMax:', averageExcludingMinMax([1, 2, 3, 4, 5]));

File: programming/problem_6/problem_6.swift
func averageExcludingMinMax(inputNumbers: [Int]) -> Double {
    let min = inputNumbers.min() ?? 0
    let max: Int = inputNumbers.max() ?? 0
    let filteredNumbers = inputNumbers.filter {$0 != min && $0 != max}
    let sum = filteredNumbers.reduce(0, +)
    return Double(sum) / Double(filteredNumbers.count)
}

print("averageExcludingMinMax: ", averageExcludingMinMax(inputNumbers: [1,2,3,4,5]))
